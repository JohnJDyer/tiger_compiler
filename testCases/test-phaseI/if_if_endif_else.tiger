let
in
  if 1<>3+4 then
    if 1 & 2 | 3 then
      flush();
    endif;
  else
    flush();
  endif;
end