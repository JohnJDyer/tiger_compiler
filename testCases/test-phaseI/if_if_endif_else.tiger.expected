main:
	add, 3, 4, $t0
	assign, $t1, 1, 
	brneq, 1, $t0, compare_label0
	assign, $t1, 0, 
compare_label0:
	breq, $t1, 0, if_label1
	and, 1, 2, $t2
	or, $t2, 3, $t3
	breq, $t3, 0, if_label2
	call, flush
if_label2:
	goto, if_label3, , 
if_label1:
	call, flush
if_label3:
	return, , , 
